src/eval_cache.rs

```diff
@@ -177,41 +167,26 @@ impl CacheManifest {
     }
 
     /// Merge recursively the new matches with the existing entries in the cache
-    #[must_use]
-    pub fn add_results(self, matches: BTreeMap<FormatterName, BTreeMap<PathBuf, Mtime>>) -> Self {
-        // Get a copy of the old matches
-        let mut new_matches = self.matches.to_owned();
-        // This is really ugly. Get a second copy to work around lifetime issues.
-        let mut new_matches_cmp = self.matches.to_owned();
-
-        // Merge all the new matches into it
+    pub fn add_results(&mut self, matches: BTreeMap<FormatterName, BTreeMap<PathBuf, Mtime>>) {
         for (name, path_infos) in matches {
-            let mut def = BTreeMap::new();
-            let merged_path_infos = new_matches_cmp
-                .get_mut(&name)
-                .unwrap_or_else(|| def.borrow_mut());
-            for (path, mtime) in path_infos {
-                merged_path_infos.insert(path.clone(), mtime);
+            if let Some(old_path_infos) = self.matches.get_mut(&name) {
+                old_path_infos.extend(path_infos);
+            } else {
+                self.matches.insert(name, path_infos);
             }
-            new_matches.insert(name, merged_path_infos.to_owned());
-        }
-
-        Self {
-            formatters: self.formatters,
-            matches: new_matches,
         }
     }
 }
 
 /// Gets all the info we want from the formatter
-fn load_formatter_info(fmt: &Formatter) -> Result<FormatterInfo> {
-    let command = fmt.command.clone();
+fn load_formatter_info(fmt: Formatter) -> Result<FormatterInfo> {
+    let command = fmt.command;
     let command_mtime = get_path_mtime(&command)?;
     // Resolve symlinks and everything
     let command_resolved = std::fs::canonicalize(command.clone())?;
     let command_resolved_mtime = get_path_mtime(&command_resolved)?;
-    let options = fmt.options.clone();
-    let work_dir = fmt.work_dir.clone();
+    let options = fmt.options;
+    let work_dir = fmt.work_dir;
```

Comment:
Here, we can just move the values off the `Formatter` struct because it's about to drop, and reuse them for other stuff